clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -854,6 +854,17 @@ else if (enableAutoCommit)
         this.groupId = groupId;
     }
 
+    private static String buildClientId(String configuredClientId, GroupRebalanceConfig rebalanceConfig) {
+        if (!configuredClientId.isEmpty())
+            return configuredClientId;
+
+        if (rebalanceConfig.groupId != null && !rebalanceConfig.groupId.isEmpty())
+            return "consumer-" + rebalanceConfig.groupId + "-" + rebalanceConfig.groupInstanceId.orElseGet(() ->
+                    CONSUMER_CLIENT_ID_SEQUENCE.getAndIncrement() + "");
```

Comment:
nit: do we need the "" in the end for conversion? I think Java is smart enough without it.

Reply:
It doesn't compile without it because the supplier passed to `orElseGet` must have type `Supplier<? extends String>`.